### PR TITLE
Port tests to munit

### PR DESCRIPTION
**Pull request by cquiroz**
This PR ports all tests to munit. I had to disable a few tests that I'll point in the comments.
This also upgrades discipline-core

### CODE REVIEW

**Comment by julien-truffaut** [comment]
Looks great. I didn't see any test disabled .

File: test/shared/src/test/scala/monocle/std/StringsSpec.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ class StringsSpec extends MonocleSuite {
   checkAll("String to Int", PrismTests(stringToInt))
   checkAll("String to Long", PrismTests(stringToLong))
   checkAll("String to UUID", PrismTests(stringToUUID))
-  checkAll("String to URI", PrismTests(stringToURI))
+  // checkAll("String to URI", PrismTests(stringToURI))
```
Comment (julien-truffaut) [inline test/shared/src/test/scala/monocle/std/StringsSpec.scala:23]: Here is one.

File: example/src/test/scala/other/ImportExample.scala [hunk 2]
```diff
@@ -39,16 +35,16 @@ class ImportExample extends AnyFunSuite with Configuration with FunSuiteDiscipli
...
-    each[List[Int], Int].modify(_ + 1)(List(1, 2, 3)) shouldEqual List(2, 3, 4)
+    assertEquals(each[List[Int], Int].modify(_ + 1)(List(1, 2, 3)), List(2, 3, 4))
   }
 
   test("monocle.std.all._ brings all polymorphic Optic instances in scope for standard Scala classes") {
     import monocle.function.all._
 
     // do not compile because Head instance for HList is not in scope
-    illTyped("""head[Int :: HNil, Int].modify(1 :: HNil, _ + 1) shouldEqual (2 :: HNil)""")
+    illTyped("""assertEquals(head[Int :: HNil, Int].modify(1 :: HNil, _ + 1),  (2 :: HNil))""")
```
Comment (julien-truffaut) [inline example/src/test/scala/other/ImportExample.scala:45]: On a second pass, we may want to use `compileErrors` from munit instead of illTyped

**Comment by julien-truffaut** [comment]
I propose that we merge it and a few tickets to tackle the commented tests. WDYT?

**Review by julien-truffaut — APPROVED** [review]

**Comment by cquiroz** [comment]
Sounds good, I think munit will make it easier to support dotty

**Comment by julien-truffaut** [comment]
> Sounds good, I think munit will make it easier to support dotty

Definitely!

**Pull request merged**